### PR TITLE
Add http error identification for credentials and hostname errors

### PR DESCRIPTION
**Pull request by SanjeevGitProfile**
Internal ticket: https://jira.cloudera.com/browse/DBT-372

The error printed seems to be generic, so adding
code to identity the failure from Hive connection.

Test plan:
  * Move into a dbt hive project and install dbt-hive from this git repo
  * Add profile details in your home dbt file like below, 
  dbt_hive_demo:
    outputs:
      dev_cia_cdp:
        auth_type: ldap
        host: {}
        http_path: {}
        password: {}
        port: 443
        schema: dbt_hive_demo
        threads: 4
        type: hive
        use_http_transport: true
        use_ssl: true
        user: {}
    target: dev_cia_cdp
  * Update with wrong hostname or wrong user credentials
  * run > dbt debug, and you should see the type of error in the console
  * Add below snowplow configs to your .env if you face any snowplow error SNOWPLOW_ENDPOINT=https://dcevents.cldrteam.datacoral.io/cldrteam/apievents
SNOWPLOW_TIMEOUT=10
SNOWPLOW_API_KEY=<your_key>
SNOWPLOW_ENNV=prod

Signed-off-by: Sanjeev kumar N <sanjeevitcit@gmail.com>

### CODE REVIEW

**Comment by SanjeevGitProfile** [comment]
Hive host connection error
hive server error
Hive credentials error
hive credential error
Passed on valid profile
dbt hive pass

File: dbt/adapters/hive/connections.py [hunk 1]
```diff
@@ -226,6 +228,10 @@ def open(cls, connection):
     def exception_handler(self, sql: str):
         try:
             yield
+        except HttpError as httpError:
+            print ("HTTP Authorization error: " + str(httpError))
```
Comment (TapasSenapati) [inline dbt/adapters/hive/connections.py:232]: The httperror 'cannot unpack non-iterable object' is not very user friendly here.  
Can we just pass that to log for our debugging and  for user just print a friendly message like "HTTP  401 Authorization error: Please check you have entered correct credentials)

File: dbt/adapters/hive/connections.py [hunk 2]
```diff
@@ -226,6 +228,10 @@ def open(cls, connection):
...
+        except HiveServer2Error as hiveError:
+            print("Hive server connection error: " + str(hiveError))
```
Comment (TapasSenapati) [inline dbt/adapters/hive/connections.py:234]: Same here.

**Comment by SanjeevGitProfile** [comment]
Updated logs for error message:
Please note type of exception caught for sql error and hive hostname is same, so I tried keep that message generic.

creds error

Hive server error

sql error

dbt pass

**Comment by tovganesh** [comment]
Looks good to me @SanjeevGitProfile please check if the conflict resolution is correct.

**Review by tovganesh — APPROVED** [review]
looks ok to me. 
@SanjeevGitProfile please check if the conflict merge is correct before merging.

**Pull request merged**